tapdb/universe_federation.go

```diff
@@ -174,4 +211,142 @@ func (u *UniverseFederationDB) LogNewSyncs(ctx context.Context,
 	})
 }
 
-var _ universe.FederationLog = (*UniverseFederationDB)(nil)
+// UpsertFederationSyncConfig upserts both the general and universe specific
+// federation sync configs.
+func (u *UniverseFederationDB) UpsertFederationSyncConfig(
+	ctx context.Context, generalSyncConfig *universe.FedGeneralSyncConfig,
+	uniSyncConfigs []*universe.FedUniSyncConfig) error {
```

Comment:
Can use var args here and below if you want to save a few lines when you map from the gRPC proto to this struct.